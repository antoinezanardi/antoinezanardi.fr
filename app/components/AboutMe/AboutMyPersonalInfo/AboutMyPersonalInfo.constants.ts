const ENTJ_PERSONALITY_HREF = "https://www.16personalities.com/fr/la-personnalite-entj";

export {
  ENTJ_PERSONALITY_HREF,
};