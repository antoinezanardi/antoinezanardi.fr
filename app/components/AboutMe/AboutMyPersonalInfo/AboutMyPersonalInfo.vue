<template>
  <div>
    <div class="card-body">
      <h3 class="h3 mb-2 mt-0 title">
        <WrappedFontAwesomeIcon
          classes="me-2"
          icon="fa-id-card"
        />

        <span>
          {{ $t('AboutMyPersonalInfo.personalInfo') }}
        </span>
      </h3>

      <hr>

      <div class="row">
        <div class="col-sm-4">
          <b
            class="text-uppercase"
          >
            {{ `${$t('AboutMyPersonalInfo.genre')} :` }}
          </b>
        </div>

        <div
          id="genre"
          class="col-sm-8"
        >
          {{ $t('AboutMyPersonalInfo.male') }}
        </div>
      </div>

      <div class="mt-3 row">
        <div class="col-sm-4">
          <b
            class="text-uppercase"
          >
            {{ `${$t('AboutMyPersonalInfo.age')} :` }}
          </b>
        </div>

        <div
          id="age"
          class="col-sm-8"
        >
          {{ $t('AboutMyPersonalInfo.yearsOld', { age }) }}
        </div>
      </div>

      <div class="mt-3 row">
        <div class="col-sm-4">
          <b
            class="text-uppercase"
          >
            {{ `${$t('AboutMyPersonalInfo.email')} :` }}
          </b>
        </div>

        <div class="col-sm-8">
          <b>
            <a
              :aria-label="$t('AboutMyPersonalInfo.emailLink', { 'email': config.public.email })"
              :href="`mailto:${config.public.email}`"
            >
              {{ config.public.email }}
            </a>
          </b>
        </div>
      </div>

      <div class="mt-3 row">
        <div class="col-sm-4">
          <b
            class="text-uppercase"
          >
            {{ `${$t('AboutMyPersonalInfo.phone')} :` }}
          </b>
        </div>

        <div class="col-sm-8">
          <b>
            <a
              id="phone-number"
              :aria-label="$t('AboutMyPersonalInfo.phoneLink', { 'phone': formattedPhoneNumber })"
              :href="`tel:${config.public.phoneNumber}`"
            >
              {{ formattedPhoneNumber }}
            </a>
          </b>
        </div>
      </div>

      <div class="mt-3 row">
        <div class="col-sm-4">
          <b
            class="text-uppercase"
          >
            {{ `${$t('AboutMyPersonalInfo.address')} :` }}
          </b>
        </div>

        <div
          id="address"
          class="col-sm-8"
        >
          {{ config.public.address }}
        </div>
      </div>

      <div class="mt-3 row">
        <div class="col-sm-4">
          <b
            class="text-uppercase"
          >
            {{ `${$t('AboutMyPersonalInfo.languages')} :` }}
          </b>
        </div>

        <div
          class="col-sm-8"
        >
          {{ $t('AboutMyPersonalInfo.englishAndFrench') }}
        </div>
      </div>

      <div class="align-items-center mt-3 row">
        <div class="col-sm-4">
          <b
            class="text-uppercase"
          >
            {{ `${$t('AboutMyPersonalInfo.licenses')} :` }}
          </b>
        </div>

        <div class="col-sm-8 d-flex">
          <WrappedFontAwesomeIcon
            :aria-label="$t('AboutMyPersonalInfo.car')"
            class="me-2"
            data-bs-toggle="tooltip"
            icon="fa-car"
            role="img"
            size="2x"
            :title="$t('AboutMyPersonalInfo.car')"
          />

          <WrappedFontAwesomeIcon
            :aria-label="$t('AboutMyPersonalInfo.boat')"
            class="me-2"
            data-bs-toggle="tooltip"
            icon="fa-anchor"
            role="img"
            size="2x"
            :title="$t('AboutMyPersonalInfo.boat')"
          />
        </div>
      </div>

      <div class="align-items-center mt-3 row">
        <div class="col-sm-4">
          <b
            class="text-uppercase"
          >
            {{ `${$t('AboutMyPersonalInfo.workingAt')} :` }}
          </b>
        </div>

        <div class="col-sm-8">
          <a
            :aria-label="$t('AboutMyPersonalInfo.companyLink', { 'company': 'Daveo' })"
            href="https://www.daveo.fr/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <NuxtImg
              alt="Daveo"
              class="daveo-logo"
              src="/images/logos/daveo-logo.webp"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import WrappedFontAwesomeIcon from "~/components/shared/Icons/WrappedFontAwesomeIcon/WrappedFontAwesomeIcon.vue";

const config = useRuntimeConfig();

const age = computed<number>(() => {
  const birthday = new Date("1996-04-14");
  const epochYear = 1970;

  return new Date(Date.now() - birthday.getTime()).getUTCFullYear() - epochYear;
});

const formattedPhoneNumber = computed<string>(() => {
  const matches = config.public.phoneNumber.match(/.{2}/gu);

  return matches ? matches.join(" ") : "?";
});
</script>

<style lang="scss" scoped>
.daveo-logo {
  margin-left: -5px;
  width: 150px;
}
</style>