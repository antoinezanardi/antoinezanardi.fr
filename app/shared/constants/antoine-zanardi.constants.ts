const ANTOINE_ZANARDI_FULL_NAME = "Antoine ZANARDI";

const ANTOINE_ZANARDI_BIRTH_DATE = "1996-04-14";

const ANTOINE_ZANARDI_GITHUB_URL = "https://github.com/antoinezanardi/";

const ANTOINE_ZANARDI_GITHUB_AVATAR_URL = "https://avatars.githubusercontent.com/u/11405981?v=4";

const ANTOINE_ZANARDI_LINKEDIN_URL = "https://www.linkedin.com/in/antoinezanardi/";

export {
  ANTOINE_ZANARDI_FULL_NAME,
  ANTOINE_ZANARDI_BIRTH_DATE,
  ANTOINE_ZANARDI_GITHUB_URL,
  ANTOINE_ZANARDI_GITHUB_AVATAR_URL,
  ANTOINE_ZANARDI_LINKEDIN_URL,
};