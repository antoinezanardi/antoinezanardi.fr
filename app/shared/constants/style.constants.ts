const PRIMARY_COLOR = "#12671D";

const BLACK_COLOR = "#000000";

const WHITE_COLOR = "#FFFFFF";

export {
  PRIMARY_COLOR,
  BLACK_COLOR,
  WHITE_COLOR,
};