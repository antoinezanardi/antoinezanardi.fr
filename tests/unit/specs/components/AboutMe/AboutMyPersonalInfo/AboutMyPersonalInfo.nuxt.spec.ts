import { mockNuxtImport } from "@nuxt/test-utils/runtime";
import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { TEST_NUXT_RUNTIME_CONFIG } from "@tests/unit/utils/constants/nuxt-test.constants";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

import AboutMyPersonalInfo from "@/components/AboutMe/AboutMyPersonalInfo/AboutMyPersonalInfo.vue";

describe("AboutMyPersonalInfo Component", () => {
  let wrapper: ReturnType<typeof mount<typeof AboutMyPersonalInfo>>;

  mockNuxtImport("useRuntimeConfig", () => () => TEST_NUXT_RUNTIME_CONFIG);

  async function mountAboutMyPersonalInfoComponent(options: ComponentMountingOptions<typeof AboutMyPersonalInfo> = {}):
  Promise<ReturnType<typeof mount<typeof AboutMyPersonalInfo>>> {
    return mountSuspendedComponent(AboutMyPersonalInfo, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountAboutMyPersonalInfoComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Age", () => {
    it("should render age when rendered.", () => {
      const age = wrapper.find<HTMLDivElement>("#age");

      expect(age.text()).toBe("25 ans");
    });
  });

  describe("Phone Number", () => {
    it("should render formatted phone number when rendered.", () => {
      const phoneNumber = wrapper.find<HTMLDivElement>("#phone-number");

      expect(phoneNumber.text()).toBe("12 34 56 78 90");
    });

    it.skip("should render ? when phone number in runtime config doesn't have a correct format.", async() => {
      wrapper = await mountAboutMyPersonalInfoComponent();
      const phoneNumber = wrapper.find<HTMLDivElement>("#phone-number");
      // mockNuxtImport("useRuntimeConfig", () => () => ({
      //   public: {
      //     phoneNumber: "mdr",
      //   },
      // }));

      expect(phoneNumber.text()).toBe("?");
    });
  });

  describe("Address", () => {
    it("should render address when rendered.", () => {
      const address = wrapper.find<HTMLDivElement>("#address");

      expect(address.text()).toBe("1234 Elm Street");
    });
  });
});