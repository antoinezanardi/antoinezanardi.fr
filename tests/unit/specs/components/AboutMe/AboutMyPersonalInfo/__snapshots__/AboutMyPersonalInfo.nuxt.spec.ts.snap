// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutMyPersonalInfo Component > should match snapshot when rendered. 1`] = `
"<div data-v-b7a042bc="">
  <div data-v-b7a042bc="" class="card-body">
    <h3 data-v-b7a042bc="" class="h3 mb-2 mt-0 title">
      <wrapped-font-awesome-icon-stub data-v-b7a042bc="" icon="fa-id-card" iconcolor="#000000" classes="me-2"></wrapped-font-awesome-icon-stub><span data-v-b7a042bc="">Informations personnelles</span>
    </h3>
    <hr data-v-b7a042bc="">
    <div data-v-b7a042bc="" class="row">
      <div data-v-b7a042bc="" class="col-sm-4"><b data-v-b7a042bc="" class="text-uppercase">Sexe :</b></div>
      <div data-v-b7a042bc="" id="genre" class="col-sm-8">Homme</div>
    </div>
    <div data-v-b7a042bc="" class="mt-3 row">
      <div data-v-b7a042bc="" class="col-sm-4"><b data-v-b7a042bc="" class="text-uppercase">Age :</b></div>
      <div data-v-b7a042bc="" id="age" class="col-sm-8">25 ans</div>
    </div>
    <div data-v-b7a042bc="" class="mt-3 row">
      <div data-v-b7a042bc="" class="col-sm-4"><b data-v-b7a042bc="" class="text-uppercase">email :</b></div>
      <div data-v-b7a042bc="" class="col-sm-8"><b data-v-b7a042bc=""><a data-v-b7a042bc="" aria-label="Envoyez-moi un email à john@doe.com" href="mailto:john@doe.com">john@doe.com</a></b></div>
    </div>
    <div data-v-b7a042bc="" class="mt-3 row">
      <div data-v-b7a042bc="" class="col-sm-4"><b data-v-b7a042bc="" class="text-uppercase">Téléphone :</b></div>
      <div data-v-b7a042bc="" class="col-sm-8"><b data-v-b7a042bc=""><a data-v-b7a042bc="" id="phone-number" aria-label="Appelez-moi au 01 23 45 67 89" href="tel:0123456789">01 23 45 67 89</a></b></div>
    </div>
    <div data-v-b7a042bc="" class="mt-3 row">
      <div data-v-b7a042bc="" class="col-sm-4"><b data-v-b7a042bc="" class="text-uppercase">Adresse postale :</b></div>
      <div data-v-b7a042bc="" id="address" class="col-sm-8">1234 Elm Street</div>
    </div>
    <div data-v-b7a042bc="" class="mt-3 row">
      <div data-v-b7a042bc="" class="col-sm-4"><b data-v-b7a042bc="" class="text-uppercase">Langues :</b></div>
      <div data-v-b7a042bc="" class="col-sm-8">Anglais et Français</div>
    </div>
    <div data-v-b7a042bc="" class="align-items-center mt-3 row">
      <div data-v-b7a042bc="" class="col-sm-4"><b data-v-b7a042bc="" class="text-uppercase">Permis :</b></div>
      <div data-v-b7a042bc="" class="col-sm-8 d-flex">
        <wrapped-font-awesome-icon-stub data-v-b7a042bc="" icon="fa-car" iconcolor="#000000" size="2x" classes="" aria-label="Permis B" class="me-2" data-bs-toggle="tooltip" role="img" title="Permis B"></wrapped-font-awesome-icon-stub>
        <wrapped-font-awesome-icon-stub data-v-b7a042bc="" icon="fa-anchor" iconcolor="#000000" size="2x" classes="" aria-label="Permis Bateau Côtier" class="me-2" data-bs-toggle="tooltip" role="img" title="Permis Bateau Côtier"></wrapped-font-awesome-icon-stub>
      </div>
    </div>
    <div data-v-b7a042bc="" class="align-items-center mt-3 row">
      <div data-v-b7a042bc="" class="col-sm-4"><b data-v-b7a042bc="" class="text-uppercase">Travaille chez :</b></div>
      <div data-v-b7a042bc="" class="col-sm-8"><a data-v-b7a042bc="" aria-label="Visitez le site de Daveo" href="https://www.daveo.fr/" rel="noopener noreferrer" target="_blank">
          <nuxt-img-stub data-v-b7a042bc="" alt="Daveo" loading="lazy" src="/images/logos/daveo-logo.webp" preload="false" crossorigin="false" placeholder="false" custom="false" class="daveo-logo"></nuxt-img-stub>
        </a></div>
    </div>
  </div>
</div>"
`;
